samples/metrics_monitor/src/cttmetrics_i915_pmu.cpp

```diff
@@ -518,13 +529,35 @@ cttStatus CTTMetrics_PMU_GetValue(unsigned int count, float* out_metric_values)
             group = &g_ctx.pm.groups[pm_metric_idx];
 
             value = perf_elapsed(&group->metrics[I915_SAMPLE_BUSY]);
+            time = (double)perf_elapsed_time(&group->metrics[I915_SAMPLE_BUSY]);
+
+            double value_sema, value_wait;
+            double time_sema, time_wait;
+            bool is_sema_wait_available = is_engine_config(group->metrics[I915_SAMPLE_BUSY].config) && (3 == group->num_metrics);
+
+            if (is_sema_wait_available) {
+                value_sema = perf_elapsed(&group->metrics[I915_SAMPLE_SEMA]);
+                time_sema = (double)perf_elapsed_time(&group->metrics[I915_SAMPLE_SEMA]);
+
+                value_wait = perf_elapsed(&group->metrics[I915_SAMPLE_WAIT]);
+                time_wait = (double)perf_elapsed_time(&group->metrics[I915_SAMPLE_WAIT]);
+            }
+
             if (g_ctx.metrics[metric_idx] == CTT_AVG_GT_FREQ) {
                 value *= 1000000000; // to compensate time in ns
             }
             else {
                 value *= 100; // we need %
+                value_sema *= 100;
```

Comment:
Well, here it is possible that ```is_sema_wait_available == false```, so ```value_sema``` and ```value_wait``` will be multiple of garbage by 100. It is not a big issue in current code, but I think better to add zero initialization at declaration to avoid unpredictable values in such case

Reply:
looks done. Reopen if needed